### ISSUE TITLE
Not possible to add photos in Signal when iOS "limited access" is enabled

### ISSUE THREAD

**Issue by itsamemarkus** [signalapp/Signal-iOS#5830]
### Bug description
When having limited access for Signal enabled, you can not select new photos that Signal should have access to within the Signal app. You have to enter privacy settings each time.

![Screenshot 2024-07-03 at 19 24 09](https://github.com/signalapp/Signal-iOS/assets/34944988/cb5b8b44-25c2-4c7c-bc33-e1e6a604e213)

Technical details about Limited Access:

https://developer.apple.com/documentation/photokit/delivering_an_enhanced_privacy_experience_in_your_photos_app

### Steps to reproduce
- only allow limited access for photos when Signal asks for it initially (or enable limited access via privacy settings later)
- Create new message
- Photos button

**Actual result:** no photos (or only the manually selected ones via settings) are displayed and there is no option to add photos.

**Expected result:** Signal shows system prompt with photos to select that Signal should have access to. This is example on how this looks with Strava

![Screenshot 2024-07-03 at 19 27 55](https://github.com/signalapp/Signal-iOS/assets/34944988/7fb9af9d-a6b7-4137-93c8-b477b64989b7)

I think it would be nice if this would work more easily with a privacy centric app.

### Device info

**Device**: iPhone 15 Pro

**iOS version**: 17.5.1

**Signal version:** 7.17 (171)

**Comment by psawa** [comment]
I agree, I think enabling private access would be a huge improvement. Is there any plan to implement this?

**Comment by seanwertheim** [comment]
After pressing the Plus button from within a chat, you can scroll to the right in the view that shows the photos, tap the "Manage" button, and add more photos you want Signal to have access to there, then send the photo.

**Comment by jamesdrake101** [comment]
I don't understand why someone might want anything other than "Private access", which is basically a photo/media picker kindly provided by the OS. Perfect user experience, perfect privacy. Total no brainer.

"Limited access" is laborious and awkward, "Full access" is just too intrusive.

Whatsapp supports all photo access modes.

**Comment by zoltrons** [comment]
I agree with @jamesdrake101. Whatsapp has this feature and works perfectly fine.

**Comment by antijingoist** [comment]
Requiring photos permissions on iOS is optional for accessing photos and saving photos. Expected Signal would work with Photo Library Access set to None.

The little carousel is cute, but I'd prefer not having that feature, and instead using the built-in iOS methods of accessing and saving media to ensure only what I want accessed is accessed.

**Comment by AmeliaHuang325** [comment]
Cannot agree more. It is not convenient to select new images and send them each time.

**Comment by sashaweiss-signal** [comment]
Hi, we've made significant improvements in this space recently, such that it's a lot easier to adjust the limited access for the recent-photo strip as well as to pick photos using the modal picker.

I'm going to close this issue as a result. Thanks as always for being Signal users!

**Comment by itsamemarkus** [comment]
I really appreciate the efforts. It got way better than before. Especially having the `Manage` button very accessible. 

But I still don't get why it's there at all. The photo picker on iOS with private access enabled it very good out of the box. The Signal photo selection is very limited, just a small part of the screen to scroll through recent photos (which aren't there with `Limited Access` enabled.

With the native iOS picker you can scroll through all your pictures and easily go to your collections and favorites.

With the current Signal implementation there are unnecessary steps:

**Native picker:**

- `+`
- Photos
- select photo

**Signal picker**

- `+`
- Manage
- Select More Photos
- Select photo
- Done
- Select photo in Signal

What's wrong with this? (Except the pictures of my mouse cable)

![Image](https://github.com/user-attachments/assets/fb4893cf-6167-4418-9883-052f7736fee8)

**Comment by haileys** [comment]
> Hi, we've made significant improvements in this space recently, such that it's a lot easier to adjust the limited access for the recent-photo strip as well as to pick photos using the modal picker.

Thanks @sashaweiss-signal, this is a good improvement.

Still, ideally Signal should support the native iOS photo picker imo. There's fewer taps and it's more convenient to attach photos this way.

Check out the Toot! app (a mastodon client for iOS) for a good example of how well the native OS photo picker can work.

As a privacy first app, Signal should be setting the bar for other apps and raising user expectations. Privacy _doesn't_ need to be less convenient than the alternatives, but right now it is, and it's a missed opportunity.

**Comment by paulmillr** [comment]
@sashaweiss-signal it's been ~6 months. Did the change get rolled back? I still see useless signal image picker which forces to do 5 additional taps instead of none if a native was used.

Photo attached. imessage got a proper "native thumbnails" image picker, without all this mess.

![Image](https://github.com/user-attachments/assets/6e3cfe56-683d-4c6c-baf5-64d1b5899194)

**Comment by zoltrons** [comment]
> @sashaweiss-signal it's been ~6 months. Did the change get rolled back? I still see useless signal image picker which forces to do 5 additional taps instead of none if a native was used.
> 
> Photo attached. imessage got a proper "native thumbnails" image picker, without all this mess.
> 
> ![Image](https://github.com/user-attachments/assets/6e3cfe56-683d-4c6c-baf5-64d1b5899194)

No, it still on the app. You have to press Photos to open native image picker.

**Comment by paulmillr** [comment]
@zoltrons well it’s still useless. Why require 7 taps when 2 will do?

Signal is privacy-friendly. It’s expected it would be also friendly to limited photo library access.

The guide here describes in depth how to use privacy-friendly carousel properly. https://developer.apple.com/videos/play/wwdc2023/10107/
